sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -271,6 +274,54 @@ case class LoadData(
 }
 
 /**
+ * A command to truncate table.
+ *
+ * The syntax of this command is:
+ * {{{
+ *  TRUNCATE TABLE tablename [PARTITION (partcol1=val1, partcol2=val2 ...)]
+ * }}}
+ */
+case class TruncateTable(
+    tableName: TableIdentifier,
+    partitionSpec: Option[TablePartitionSpec]) extends RunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    if (!catalog.tableExists(tableName)) {
+      throw new AnalysisException(s"table in TRUNCATE TABLE does not exist: '$tableName'")
```

Comment:
In `DropTable` (which is more drastic IMO) we do not throw an `AnalysisException`...